pkg/name/name.go

```diff
@@ -390,3 +390,75 @@ func Set(val, out interface{}) error {
 	reflect.ValueOf(out).Set(reflect.ValueOf(val))
 	return nil
 }
+
+// CreatePatchObjectFromPath constructs patch object for node with path, returns nil object and error if the path is invalid.
+// eg. node:
+//     - name: NEW_VAR
+//       value: new_value
+// and path:
+//       spec.template.spec.containers.[name:discovery].env
+//     will constructs the following patch object:
+//       spec:
+//         template:
+//           spec:
+//             containers:
+//             - name: discovery
+//               env:
+//               - name: NEW_VAR
+//                 value: new_value
+func CreatePatchObjectFromPath(node interface{}, path util.Path) (map[string]interface{}, error) {
+	if len(path) == 0 {
+		return nil, fmt.Errorf("empty path %s", path)
+	}
+	if util.IsKVPathElement(path[0]) {
+		return nil, fmt.Errorf("path %s has an unexpected first element %s", path, path[0])
+	}
+	length := len(path)
+	if util.IsKVPathElement(path[length-1]) {
+		return nil, fmt.Errorf("path %s has an unexpected last element %s", path, path[length-1])
+	}
+
+	patchObj := make(map[string]interface{})
+	var currentNode, nextNode interface{}
+	nextNode = patchObj
+	for i, pe := range path {
+		currentNode = nextNode
+		// last path element
+		if i == length-1 {
+			currentNode, ok := currentNode.(map[string]interface{})
+			if !ok {
+				return nil, fmt.Errorf("path %s has an unexpected non KV element %s", path, pe)
+			}
+			currentNode[pe] = node
+			break
+		}
+
+		// if current path element is kc element, then append new item to parent list.
+		if util.IsKVPathElement(pe) {
```

Comment:
Somehow I didn't see the part in the code that returns in PathKV returns error (which it will it the path is not found). I think I was just confused by the comment above about appending to the parent list. We expect to find the item already in the parent, if not, then fail. If you agree, could you pleasure update the comment? And the check you added is unnecessary. Sorry for the confusion!

Reply:
hmm, got your point. The list item with strategic merge key is actually inserted into parent here, and it should returns error if PV is not valid; Also the parent slice is created with empty content in the previous iteration, see: https://github.com/istio/operator/blob/94630121fb028683e46c29714447b4e8a9731516/pkg/name/name.go#L459-L460

Reply:
done, removed the confusing comment.